agents/go-agents/core/auth/auth.go

```diff
@@ -111,6 +112,12 @@ func (handler handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	token := req.URL.Query().Get("token")
+	if token == "" {
+	  header := req.Header.Get("Authorization")
+	  if header != "" && strings.HasPrefix(strings.ToLower(header), "bearer") {
```

Comment:
it's impossible in our authentication implementation but still, you check `bearer` word and cut `bearer ` what if there will be a token that starts with `bearer` that is not type but just part of token like `bearer9SUcJXjsdenx`. Maybe it would be better to check if token is prefixed with `bearer `

Reply:
`Bearer` word indicates type of token, there also can be `Basic` or `Digest` which is not supported. So we cut it away to have pure token. Or i understood your question wrong ?

Reply:
Markdown helps me to confuse you =)
Let me rephrase, you check bearer word without space and cut bearer with space (7 characters) Maybe it would be better to check if token is prefixed with **bearer ** with space.

Reply:
Ok